src/main/scala/is/hail/expr/HailRep.scala

```diff
@@ -3,7 +3,7 @@ package is.hail.expr
 import is.hail.utils.Interval
 import is.hail.variant.{AltAllele, Call, Genotype, Locus, Variant}
 
-trait HailRep[T] { self =>
+trait HailRep[+T] { self =>
```

Comment:
I'm not sure whether it's better to do this or to have an `AnyHr(typ: Type) extends HailRep[Any]` class